contributors/design-proposals/api-machinery/admission-control-webhooks.md

```diff
@@ -278,6 +282,34 @@ cluster can boot from a cold start) is made more difficult by having webhooks,
 which are a dependency of the control plane. This is covered in its [own design
 doc](./admission-webhook-bootstrapping.md).
 
+## Upgrading the control plane
+
+There are two categories of webhooks: security critical (e.g., scan images for
+vulnerabilities) and nice-to-have (set labels).
+
+Security critical webhooks cannot work with Kubernetes types they don't have
```

Comment:
Doesn't have to be here, but at some point we should document/explain the decode->encode->semantic-equals check that can be used inside of a webhook to fail closed on a new field that is actually used (not a pre-check).

Reply:
Yeah, not here, because I actually don't want to tell people to do that. I think it adds a lot of confusion for what amounts to only an 80% solution (e.g., doesn't detect annotation changes or semantic meaning changes; neither should happen but I'll be astonished if we live up to that ideal 100%).